File: packages/python-sdk/src/treedx_sdk/adapters/context.py
from __future__ import annotations

from typing import Any

from .common import json_request, segment
from treedx_sdk.transport import Transport


class ContextAdapter:
    def __init__(self, transport: Transport) -> None:
        self.transport = transport

    def build(self, repo_id: str, body: Any) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/context/build", body)

    def parse(self, repo_id: str, body: Any) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/context/parse-ctx", body)

File: packages/ts-sdk/src/treedx/adapters/context.ts
import { jsonRequest, segment, type TreeDxAdapterContext } from './common.js';

export class ContextAdapter {
  constructor(private readonly context: TreeDxAdapterContext) {}
  build(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/context/build`, input); }
  parse(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/context/parse-ctx`, input); }
}
